core/src/main/java/com/zextras/carbonio/files/message_broker/consumers/KvChangedConsumer.java

```diff
@@ -0,0 +1,97 @@
+// SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package com.zextras.carbonio.files.message_broker.consumers;
+
+import com.zextras.carbonio.files.dal.dao.ebean.FileVersion;
+import com.zextras.carbonio.files.dal.dao.ebean.Node;
+import com.zextras.carbonio.files.dal.repositories.interfaces.FileVersionRepository;
+import com.zextras.carbonio.files.dal.repositories.interfaces.NodeRepository;
+import com.zextras.carbonio.message_broker.config.EventConfig;
+import com.zextras.carbonio.message_broker.consumer.BaseConsumer;
+import com.zextras.carbonio.message_broker.events.generic.BaseEvent;
+import com.zextras.carbonio.message_broker.events.services.service_discover.KvChanged;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class KvChangedConsumer extends BaseConsumer {
+
+  private static final Logger logger = LoggerFactory.getLogger(KvChangedConsumer.class);
+
+  private final NodeRepository nodeRepository;
+  private final FileVersionRepository fileVersionRepository;
+
+  public KvChangedConsumer(NodeRepository nodeRepository, FileVersionRepository fileVersionRepository) {
+    this.nodeRepository = nodeRepository;
+    this.fileVersionRepository = fileVersionRepository;
+  }
+
+  @Override
+  protected EventConfig getEventConfig() {
+    return EventConfig.KV_CHANGED;
+  }
+
+  @Override
+  public void doHandle(BaseEvent baseMessageBrokerEvent) {
+    KvChanged kvChanged = (KvChanged) baseMessageBrokerEvent;
+    logger.info("Received KvChanged({}, {})", kvChanged.getKey(), kvChanged.getValue());
+
+    // switch for future use cases with KVs
+    switch (kvChanged.getKey()) {
+      case "carbonio-files/max-number-of-versions":
+        handleMaxVersionNumerChanged(Integer.parseInt(kvChanged.getValue()));
+        break;
+      default:
+        logger.info("Event was ignored");
+        break;
```

Comment:
I like the idea but I would use the more compact format:

```suggestion
    switch (kvChanged.getKey()) {
      case "carbonio-files/max-number-of-versions" -> handleMaxVersionNumerChanged(Integer.parseInt(kvChanged.getValue()));
      default -> logger.info("Event was ignored");
```